### PR TITLE
feat(route): add twitch

### LINKED ISSUE #12883: 请求添加 Twitch 的 RSS
[Issuehunt badges](https://oss.issuehunt.io/r/DIYgod/RSSHub/issues/12883)

### 类型

音视频

### 网站地址

https://www.twitch.tv/

### 网站描述

一个直播平台

### 需要生成什么内容？

类似现有的哔哩哔哩用户动态一样，只要订阅了该用户后。可以自动获取以下内容，也可以选择性订阅以下内容。
1. 支持订阅用户的开播提醒；
2. 直播时间表推送；
3. 用户上传的视频

### 额外描述

_No response_

### 这不是重复的 RSS 请求

- [X] 我已经搜索了[现有 issue](https://github.com/DIYgod/RSSHub/issues) 和 [pull requests](https://github.com/DIYgod/RSSHub/pulls)，以确保该 RSS 尚未被请求。

---

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>

### Backers (Total: $10.00)

- [vstory vstory](https://oss.issuehunt.io/u/vstory) ($10.00)

### Submitted pull Requests
- [#13464 feat(route): add twitch](https://oss.issuehunt.io/r/DIYgod/RSSHub/pull/13464)
---

### Tips

- Checkout the [Issuehunt explorer](https://oss.issuehunt.io/r/DIYgod/RSSHub/) to discover more funded issues.
- Need some help from other developers? [Add your repositories](https://oss.issuehunt.io/r/new) on IssueHunt to raise funds.
</details>

### PR DESCRIPTION
**Pull request by hoilc**
## Involved Issue / 该 PR 相关 Issue

Close #12883

## Example for the Proposed Route(s) / 路由地址示例

```routes
/twitch/live/riotgames
/twitch/video/riotgames/highlights
/twitch/schedule/riotgames
```

## New RSS Route Checklist / 新 RSS 路由检查表
  
- [x] New Route / 新的路由
  - [x] Follows [v2 Script Standard](https://docs.rsshub.app/joinus/advanced/script-standard) / 跟随 [v2 路由规范](https://docs.rsshub.app/zh/joinus/advanced/script-standard)
- [x] Documentation / 文档说明
  - [x] EN / 英文文档
  - [x] CN / 中文文档
- [ ] Full text / 全文获取
  - [ ] Use cache / 使用缓存
- [ ] Anti-bot or rate limit / 反爬/频率限制 
  - [ ] If yes, do your code reflect this sign? / 如果有, 是否有对应的措施? 
- [x] [Date and time](https://docs.rsshub.app/joinus/advanced/pub-date) / [日期和时间](https://docs.rsshub.app/zh/joinus/advanced/pub-date)
  - [x] Parsed / 可以解析
  - [x] Correct time zone / 时区正确
- [ ] New package added / 添加了新的包
- [ ] `Puppeteer`

### CODE REVIEW

**Comment by hoilc** [comment]
/test

```routes
/twitch/channel/lolpacifictw
```

File: website/docs/routes/live.md [hunk 1]
```diff
@@ -17,6 +17,12 @@
 
 <Route author="nwindz" example="/showroom/room/93401" path="/showroom/room/:id" paramsDesc={['直播间 id, 打开浏览器控制台，刷新页面，找到请求中的room_id参数']}/>
 
+## Twitch {#twitch}
+
+### Channel {#twitch-channel}
+
+<Route author="hoilc" path="/twitch/channel/:login" example="/twitch/channel/riotgames" paramsDesc={['Twitch Channel ID']} radar="1"/>
```
Comment (TonyRL) [inline website/docs/routes/live.md:24]: I think it will be better to change it to `/twitch/live/:login`.

**Review by TonyRL — COMMENTED** [review]
#12883 also requested channel videos and stream schedule.

Channel videos:
```json
  {
    "operationName": "ChannelVideoShelvesQuery",
    "variables": {
      "channelLogin": "**change me**",
      "first": 5
    },
    "extensions": {
      "persistedQuery": {
        "version": 1,
        "sha256Hash": "7b31d8ae7274b79d169a504e3727baaaed0d5ede101f4a38fc44f34d76827903"
      }
    }
  }
```

Stream schedule:
```json
{
  "operationName": "StreamSchedule",
  "variables": {
    "login": "** change me **",
    "startingWeekday": "MONDAY",
    "utcOffsetMinutes": 480,
    "startAt": "2023-10-01T16:00:00.000Z",
    "endAt": "2023-10-08T15:59:59.059Z"
  },
  "extensions": {
    "persistedQuery": {
      "version": 1,
      "sha256Hash": "01925339777a81111ffac469430bc4ea4773c18a3c1642f1b231e61e2278ea41"
    }
  }
}
```

**Comment by hoilc** [comment]
@TonyRL Thank you, kindly review it again.

File: lib/v2/twitch/maintainer.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+module.exports = {
+    '/channel/:login': ['hoilc'],
```
Comment (TonyRL) [inline lib/v2/twitch/maintainer.js:2]:
```suggestion
    '/live/:login': ['hoilc'],
```

**Comment by Vstory** [comment]
Very cool! Thank you for your contribution.

**Pull request merged**